StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerDataSource.swift

```diff
@@ -39,7 +39,35 @@ final class LinkAccountPickerDataSourceImplementation: LinkAccountPickerDataSour
     private let apiClient: FinancialConnectionsAPIClient
     private let clientSecret: String
     private let consumerSession: ConsumerSessionData
-    let dataAccessNotice: FinancialConnectionsDataAccessNotice?
+    var dataAccessNotice: FinancialConnectionsDataAccessNotice? {
+        var selectedAccountDataAccessNotice: FinancialConnectionsDataAccessNotice?
+        if
+            let networkedAccountsResponse,
+            let returningNetworkingUserAccountPicker = networkedAccountsResponse.display?.text?.returningNetworkingUserAccountPicker,
+            !selectedAccounts.isEmpty
+        {
+            // Bank accounts can have multiple types (ex. linked account, and manual entry account).
+            //
+            // Example output:
+            // ["bctmacct", "csmrbankacct"]
+            let selectedAccountTypes = Set(selectedAccounts
+                .map({ $0.partnerAccount.id.split(separator: "_").first })
+                .compactMap({ $0 }))
+            if selectedAccountTypes.count > 1 {
+                // if user selected multiple different account types,
+                // present a special data access notice
+                selectedAccountDataAccessNotice = returningNetworkingUserAccountPicker.multipleAccountTypesSelectedDataAccessNotice
+            } else {
+                // we get here if user selected:
+                // 1) one account
+                // 2) or, multiple accounts of the same account type
+                selectedAccountDataAccessNotice = selectedAccounts.first?.accountPickerAccount.dataAccessNotice
+            }
+        }
+        return selectedAccountDataAccessNotice ?? consentDataAccessNotice
+    }
```

Comment:
[Stripe.js code for this](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/15a677fc/src/linkedAccounts/inner/components/panes/v3/ConsentPane/content/DataAccessNoticeDrawer/useDataAccessNoticeText.ts#L31-L72) (can be double-checked for equivalency)